lib/std/jsonutils.nim

```diff
@@ -128,14 +128,14 @@ macro initCaseObject(T: typedesc, fun: untyped): untyped =
         `fun`(`key2`, typedesc[`typ`])
       result.add newTree(nnkExprColonExpr, key, val)
 
-proc checkJsonImpl(cond: bool, condStr: string, msg = "") =
-  if not cond:
-    # just pick 1 exception type for simplicity; other choices would be:
-    # JsonError, JsonParser, JsonKindError
-    raise newException(ValueError, msg)
+proc raiseJsonException(condStr: string, msg: string) =
```

Comment:
```suggestion
proc raiseJsonException(condStr: string, msg: string) {.noinline.} =
```

Reply:
done, but can't the C compiler figure out on its own (and without PGO) that it's not worth inlining because it's a noreturn ?

Reply:
With --exceptions:goto it's not a noreturn.